crates/core/src/db/datastore/locking_tx_datastore/mut_tx.rs

```diff
@@ -877,7 +877,9 @@ impl MutTxId {
         // so this method needs only to check the committed state.
         if let Some(commit_table) = commit_table {
             commit_table
-                .check_unique_constraints(row, |maybe_conflict| self.tx_state.is_deleted(table_id, maybe_conflict))
+                .check_unique_constraints(&self.committed_state_write_lock.blob_store, row, |maybe_conflict| {
+                    self.tx_state.is_deleted(table_id, maybe_conflict)
+                })
```

Comment:
The error we're getting is here:

```
---- db::datastore::locking_tx_datastore::datastore::tests::test_create_index_pre_commit stdout ----
[TRACE spacetimedb::db::datastore::locking_tx_datastore::mut_tx] INDEX CREATING: age_idx for table: 4097 and col(s): [ColId(2)]
[TRACE spacetimedb::db::datastore::locking_tx_datastore::mut_tx] INDEX CREATED: 4099 for table: 4097 and col(s): [ColId(2)]
[TRACE spacetimedb::db::datastore::locking_tx_datastore::mut_tx] CONSTRAINT CREATING: ct_Foo_age_unique for table: 4097 and cols: [ColId(2)]
[TRACE spacetimedb::db::datastore::locking_tx_datastore::mut_tx] CONSTRAINT CREATED: 4099
thread 'db::datastore::locking_tx_datastore::datastore::tests::test_create_index_pre_commit' panicked at crates/table/src/table.rs:117:21:
index out of bounds: the len is 0 but the index is 0
stack backtrace:
   0: rust_begin_unwind
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panicking.rs:645:5
   1: core::panicking::panic_fmt
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/panicking.rs:72:14
   2: core::panicking::panic_bounds_check
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/panicking.rs:209:5
   3: <usize as core::slice::index::SliceIndex<[T]>>::index
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/slice/index.rs:264:10
   4: core::slice::index::<impl core::ops::index::Index<I> for [T]>::index
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/slice/index.rs:18:9
   5: <alloc::vec::Vec<T,A> as core::ops::index::Index<I>>::index
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/alloc/src/vec/mod.rs:2829:9
   6: <spacetimedb_table::pages::Pages as core::ops::index::Index<spacetimedb_table::indexes::PageIndex>>::index
             at /home/james/linux-dev/SpacetimeDB/crates/table/src/pages.rs:24:20
   7: spacetimedb_table::table::TableInner::page_and_offset
             at /home/james/linux-dev/SpacetimeDB/crates/table/src/table.rs:117:21
   8: spacetimedb_table::table::Table::is_row_present
             at /home/james/linux-dev/SpacetimeDB/crates/table/src/table.rs:1093:30
   9: spacetimedb_table::table::Table::get_row_ref
             at /home/james/linux-dev/SpacetimeDB/crates/table/src/table.rs:346:9
  10: spacetimedb_table::table::Table::check_unique_constraints
             at /home/james/linux-dev/SpacetimeDB/crates/table/src/table.rs:174:36
  11: spacetimedb_table::table::Table::insert
             at /home/james/linux-dev/SpacetimeDB/crates/table/src/table.rs:205:9
  12: spacetimedb::db::datastore::locking_tx_datastore::mut_tx::MutTxId::insert_row_internal
             at ./src/db/datastore/locking_tx_datastore/mut_tx.rs:892:15
  13: spacetimedb::db::datastore::locking_tx_datastore::mut_tx::MutTxId::insert
             at ./src/db/datastore/locking_tx_datastore/mut_tx.rs:868:9
  14: <spacetimedb::db::datastore::locking_tx_datastore::datastore::Locking as spacetimedb::db::datastore::traits::MutTxDatastore>::insert_mut_tx
             at ./src/db/datastore/locking_tx_datastore/datastore.rs:506:9
  15: spacetimedb::db::datastore::locking_tx_datastore::datastore::tests::test_create_index_pre_commit
             at ./src/db/datastore/locking_tx_datastore/datastore.rs:1779:22
```
I think I may be failing to maintain an invariant